### PR TITLE
Two new ROOT-based Python numeric modules, for use with HEP analysis code

### PR DESCRIPTION
**Pull request by kelseymh**
rootpy-1.0.1-foss-2018b-Python-3.6.6.eb is an update for Python 3.6.6 support, which ROOT hasn't had up to now.

root_numpy-4.8.0-foss-2018b-Python-3.6.6.eb is an entirely new package for EasyBuild, which provides integrated ROOT and Numpy array support for TTrees.

### CODE REVIEW

**Comment by kelseymh** [comment]
This PR (specifically the new *root_numpy* module) depends on #10377, which has passed its checks but hasn't been reviewed or merged.  Should I merge that PR's two entries onto this branch, or wait for that PR to be merged and retrigger the checks here?

**Comment by kelseymh** [comment]
Thank you, @lexming.  I don't see why we can't use numpy v1.15.0; I'll have to do the build and make sure there are no hiccups.  I chose the "latest and greatest" just because...

The dependence is actually that root_numpy uses _both_ ROOT and numpy, combining the two into a useful Python analysis tool.  It will depend on both ROOT/6.14.06-foss-2018b-Python-3.6.6.eb and Python-3.6.6-foss-2018b.eb.  If I can use numpy-1.15.0-blah-blah, then I will update this PR.

**Comment by kelseymh** [comment]
I'm not sure why those extra files, especially the numpy-1.15.0 and Cython-0.29.16, got pulled in here.  I used `eb --update-pr=10424` and listed only the one file I changed (root_numpy).  I apologize for any confusion.

**Comment by lexming** [comment]
@kelseymh you can remove any file from this PR by prepending a colon to the name of the easyconfig
```
eb --update-pr 10424 --pr-commit-msg "some commit message" :numpy-1.15.0-foss-2018b-Python-3.6.6.eb
```
`numpy` and `Cython` are already provided by `Python-3.6.6-foss-2018b.eb` so those can be safely removed from this PR.

**Comment by kelseymh** [comment]
Question: in the previous round of failed checks, it complained about "http://" in the `homepage` entry for root_numpy.  But it didn't complain about that in rootpy, and I see the same thing in _both_ rootpy EasyConfigs (0.8.0 and my new 1.0.1).  

Rootpy's Web site is only http://www.rootpy.org/.  It doesn't have an https://www.rootpy.org/ server.

Does the validation test check whether https:// works, before complaining about it?

**Comment by lexming** [comment]
> Does the validation test check whether https:// works, before complaining about it?
Yes

Thanks for updating this PR, I'll start testing it.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 2 out of 3 (2 easyconfigs in this PR)
node150.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/21ecdf82c27dc10e897b04561a59c6da for a full test report.

File: easybuild/easyconfigs/r/rootpy/rootpy-1.0.1-foss-2018b-Python-3.6.6.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+and PyTables, but a suitable interface between them and ROOT has been
+lacking. rootpy provides the interfaces and conversion mechanisms
+required to liberate your data and to take advantage of these
+alternatives if needed."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+source_urls = ['https://github.com/rootpy/%(namelower)s/archive/']
+sources = ['%(version)s.tar.gz']
+checksums = ['7d85fc8fc4ff96a833fa593be401aa4836b755757b518d23d4c00d7a105ce27e']
```
Comment (lexming) [inline easybuild/easyconfigs/r/rootpy/rootpy-1.0.1-foss-2018b-Python-3.6.6.eb:24]: I got a tarball for rootpy v1.0.1 with the following checksum. Could you verify it?
```
f7e2bec0c2fb6837c97f5060abe0a015243d1130102a0f38807f9de0f7825de5  1.0.1.tar.gz
```

Comment (kelseymh) [reply easybuild/easyconfigs/r/rootpy/rootpy-1.0.1-foss-2018b-Python-3.6.6.eb:24]:
```suggestion
checksums = ['f7e2bec0c2fb6837c97f5060abe0a015243d1130102a0f38807f9de0f7825de5']
```

**Comment by kelseymh** [comment]
@lexming Naive question:  Should checksums change on a distribution like this?  Why didn't my own builds of this .eb file (on TAMU HPRC) fail if the checksum was wrong?

**Comment by lexming** [comment]
@kelseymh you should check why that happened, is your first tarball corrupted? does its contents differ from the tarball that can be downloaded today?

**Comment by kelseymh** [comment]
To be quite honest.  I don't know where the previous checksum arose.  The "7d85fc8f" checksum was in the file I initially committed in my local EC repository.  I tried grepping for it in my other .eb files (i.e., a misplaced copy-paste), but no matches.

In the original version of rootpy-1.0.1, which I used to actually build the module for my group, the .eb file didn't have a `checksums` entry at all (the existing rootpy-0.8.0, which I copied, does not have one either).

The rootpy file which was downloaded as part of that (successful) installation, without a checksum.  I ran `shasum -a 256` on the saved sources/.../1.0.1.tar.gz file, and it came back with "your" (correct) checksum, not the one I committed.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node150.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/853bbd72bb76b083e7dda7f7844168d6 for a full test report.

File: easybuild/easyconfigs/r/root_numpy/root_numpy-4.8.0-foss-2018b-Python-3.6.6.eb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+]
+
+use_pip = True
+sanity_pip_check = True
+
+download_dep_fail = True
+dependencies = [
+    ('Python', '3.6.6'),
+    ('ROOT', '6.14.06', versionsuffix),
+]
```
Comment (lexming) [inline easybuild/easyconfigs/r/root_numpy/root_numpy-4.8.0-foss-2018b-Python-3.6.6.eb:28]:
```suggestion
dependencies = [
    ('Python', '3.6.6'),
    ('ROOT', '6.14.06', versionsuffix),
]

use_pip = True
download_dep_fail = True
sanity_pip_check = True
```

File: easybuild/easyconfigs/r/rootpy/rootpy-1.0.1-foss-2018b-Python-3.6.6.eb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+use_pip = True
+sanity_pip_check = True
+
+download_dep_fail = True
+dependencies = [
+    ('Python', '3.6.6'),
+    ('ROOT', '6.14.06', versionsuffix),
+    ('matplotlib', '3.0.0', versionsuffix),
+    ('PyTables', '3.4.4', versionsuffix),
+]
```
Comment (lexming) [inline easybuild/easyconfigs/r/rootpy/rootpy-1.0.1-foss-2018b-Python-3.6.6.eb:35]:
```suggestion
dependencies = [
    ('Python', '3.6.6'),
    ('ROOT', '6.14.06', versionsuffix),
    ('matplotlib', '3.0.0', versionsuffix),
    ('PyTables', '3.4.4', versionsuffix),
]

use_pip = True
download_dep_fail = True
sanity_pip_check = True
```

File: easybuild/easyconfigs/r/rootpy/rootpy-1.0.1-foss-2018b-Python-3.6.6.eb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+sanity_check_paths = {
+    'files': [],
```
Comment (lexming) [inline easybuild/easyconfigs/r/rootpy/rootpy-1.0.1-foss-2018b-Python-3.6.6.eb:38]:
```suggestion
    'files': ['bin/roosh', 'bin/root2hdf5', 'bin/rootpy'],
```

**Comment by kelseymh** [comment]
Thank you, @lexming .  I'm glad those were all stylistic issues, and not functional.  I'll learn all this stuff eventually :-)

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node154.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/26b0e91f46f42389cbf2e3306304640d for a full test report.

**Comment by lexming** [comment]
Going in, thanks for following up on this PR @kelseymh !

**Pull request merged**